tracks/julia/exercises/nucleotide-count/mentoring.md

```diff
@@ -0,0 +1,40 @@
+### Mentor guidance
+
+Students that use four counting variables rather than mutating a dict should be shown the dict solution or asked to come up with it on their own if there is some doubt that they knew they could do that.
+
+We shouldn't require the mutating-a-dict strategy because it's actually pretty slow.
+
+
+### Example solutions
+
+```julia
+function count_nucleotides(strand)
+    counter = Dict(base => 0 for base in "ACGT")
+    for base in strand
+        if haskey(counter, base)
+            counter[base] += 1
```

Comment:
This to me is also something I bring up with anybody who has expressed interest in getting a performant solution.
That makes 3 Dict lookups (I feel even 1 will make things too slow)

Reply:
Without context, I don't think any solution can be said to be too slow.

I think it's fine to point out how to make things more performant, but these answers are also fine, and I think they may generalise to a wider set of problems better.